src/content/1.7/modules/concepts/controllers/admin-controllers/override-decorate-controller.md

```diff
@@ -0,0 +1,171 @@
+    defaults:
+      _controller: 'MyModule\Controller\DemoController::demoAction'
+```
+
+{{% notice tip %}}
+In the above example, the `path` has not been changed however you can change it to whatever you want like `/demo/orders` ! This however will re-route internal links but external or hardcoded links will keep targeting the old path.
+{{% /notice %}}
+
+{{% notice warning %}}
+Keep the item `_legacy_controller` if your controller relies on it to configure a [AdminSecurity annotation](/1.7/development/architecture/migration-guide/controller-routing/#access-rules-convention) such as `@AdminSecurity("is_granted('read', request.get('_legacy_controller'))")`
+{{% /notice %}}
+
+{{% notice warning %}}
+Keep the items `_legacy_controller` and `_legacy_link` if you want to reroute internal links and legacy URLs like `index.php?controller=AdminOrders` as well.
+{{% /notice %}}
+
+
+Thanks to this, whenever an HTTP request is matched to the route `admin_orders_index`, then your controller `demoAction()` will be executed.
+
+All module routes are prefixed with `modules`, which is why the URL of this remapped route is `modules/demo/orders`.
+
+{{% notice tip %}}
+Routing is computed and cached by Symfony. You will probably need to clear this cache for Symfony to acknowledge your updated routing.
+You can do it by using `php bin/console cache:clear`.
+{{% /notice %}}
+
+{{% notice tip %}}
+if you have trouble writing the right routing configuration for your controller, you can use Symfony debugger to dump the routes with `php bin/console debug:router`.
+{{% /notice %}}
+
+## Override the controller
+
+Since 1.7.7, PrestaShop Back Office controllers are registered as services, and like all public services, they can be [overridden][override-services] by modules.
+
+For example, the controller responsible for the page "Improve > Design > CMS Pages" is registered with service ID `PrestaShopBundle\Controller\Admin\Improve\Design\CmsPageController`.
+
+With the following configuration item, we can override this configuration to make it target our custom module
+```yaml
+# modules/your-module/config/services.yml
+  'PrestaShopBundle\Controller\Admin\Improve\Design\CmsPageController':
+    class: MyModule\Controller\DemoController
+
+```
+
+Thanks to this, whenever Symfony forwards a request to the Core controller `PrestaShopBundle\Controller\Admin\Improve\Design\CmsPageController` it will be forwarded to `DemoController` instead.
+
+{{% notice warning %}}
+**This method is not recommended** unless you intend to rewrite the whole controller. In addition, if the implementation is updated in later versions of PrestaShop, your override will ignore these updates, which might create bugs.
+{{% /notice %}}
+
+{{% notice tip %}}
+if you have trouble writing the right service configuration for your controller, you can use Symfony debugger to dump the routes with `php bin/console debug:container`. It can also be helpful to find the service ID of the controller.
```

Comment:
```suggestion
If you have trouble writing the right service configuration for your controller, you can use Symfony debugger to dump the routes with `php bin/console debug:container`. It can also be helpful to find the service ID of the controller.
```